dev/PushNotifications/PushNotificationCreateChannelResult.cpp

```diff
@@ -0,0 +1,25 @@
+﻿#include "pch.h"
+#include "PushNotificationCreateChannelResult.h"
+#include "PushNotificationCreateChannelResult.g.cpp"
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+    PushNotificationCreateChannelResult::PushNotificationCreateChannelResult(Microsoft::ProjectReunion::PushNotificationChannel const& channel, winrt::hresult const& extendedError, Microsoft::ProjectReunion::PushNotificationChannelStatus const& status)
```

Comment:
If this fails, would the channel be nullptr in the projection? Have we tested that case? I worry because this is a pass by ref and doesn't seem like it handles null channels.

Reply:
I need to understand this question better. I think what you are suggesting is in cases like:

1. if we fail to allocate ProjectReunion channel where we are creating it in the PushNotificationManager.cpp.
2. if we fail to fetch the PushNotification channel from across the ABI due to some exception

Then in both the cases, we catch the exception to evaluate if we can retry this, if we fail to retry then we prepare channelResult with CompletedFailure status and pass nullptr.

channelResult = winrt::make<winrt::Microsoft::ProjectReunion::implementation::PushNotificationCreateChannelResult>(nullptr, ex.code(), PushNotificationChannelStatus::CompletedFailure);
 
So yes, the projection will be nullptr and the user should only access the channel projection after checking the status. Below is an example, 

auto result = sender.GetResults();
                if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
                {
                    auto channelUri = result.Channel().Uri();
                    auto channelExpiry = result.Channel().ExpirationTime();

                    //sendRequestToServer(channelUri);
                    std::cout << "channelUri: " << winrt::to_string(channelUri.ToString()) << std::endl;
                    // Persist the channelUri and Expiry in the App Service
                }
                else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
                {
                       std::cout << "Channel is nullptr" << std::endl;
                 }